Add USB Host 1 ressources + correct ID of USB Host 2

Integrated in latest kernels with different names for platform devices.
Ex: mxc_ehci1 -> mxc_usbh1

Index: linux-2.6.29.4/arch/arm/mach-mx2/devices.c
===================================================================
--- linux-2.6.29.4.orig/arch/arm/mach-mx2/devices.c	2009-06-04 11:30:50.000000000 +0200
+++ linux-2.6.29.4/arch/arm/mach-mx2/devices.c	2009-06-04 11:30:50.000000000 +0200
@@ -609,6 +609,30 @@
 	.num_resources = ARRAY_SIZE(mxc_otg_resources),
 };
 
+static struct resource mxc_ehci1_resources[] = {
+	{
+		.start = OTG_BASE_ADDR + 0x200,
+		.end   = OTG_BASE_ADDR + 0x3ff,
+		.flags = IORESOURCE_MEM,
+	}, {
+		.start = MXC_INT_USB1,
+		.flags = IORESOURCE_IRQ,
+	},
+};
+
+static u64 ehci1_dmamask = ~(u32) 0;
+
+struct platform_device mxc_ehci1 = {
+	.name = "mxc-ehci",
+	.id = 1,
+	.dev = {
+		.coherent_dma_mask = 0xffffffff,
+		.dma_mask = &ehci1_dmamask,
+	},
+	.num_resources = ARRAY_SIZE(mxc_ehci1_resources),
+	.resource = mxc_ehci1_resources,
+};
+
 static struct resource mxc_ehci2_resources[] = {
 	{
 		.start = OTG_BASE_ADDR + 0x400,
@@ -624,7 +648,7 @@
 
 struct platform_device mxc_ehci2 = {
 	.name = "mxc-ehci",
-	.id = 1,
+	.id = 2,
 	.dev = {
 		.coherent_dma_mask = 0xffffffff,
 		.dma_mask = &ehci2_dmamask,
Index: linux-2.6.29.4/arch/arm/mach-mx2/devices.h
===================================================================
--- linux-2.6.29.4.orig/arch/arm/mach-mx2/devices.h	2009-06-04 11:30:50.000000000 +0200
+++ linux-2.6.29.4/arch/arm/mach-mx2/devices.h	2009-06-04 11:30:50.000000000 +0200
@@ -28,4 +28,5 @@
 extern struct platform_device imx_ssi_device1;
 extern struct platform_device mxc_dam_device;
 extern struct platform_device mxc_otg;
+extern struct platform_device mxc_ehci1;
 extern struct platform_device mxc_ehci2;
